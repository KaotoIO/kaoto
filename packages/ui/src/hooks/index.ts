export * from './applied-schema.hook';
export * from './entities';
export * from './local-storage.hook';
export * from './schema-bridge.hook';
