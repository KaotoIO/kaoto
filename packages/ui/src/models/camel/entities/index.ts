export * from './base-entity';
export * from './kamelet-binding-overrides';
export * from './pipe-overrides';
