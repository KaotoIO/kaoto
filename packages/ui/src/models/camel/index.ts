export * from './camel-resource';
export * from './camel-route-resource';
export * from './integration-resource';
export * from './kamelet-resource';
export * from './kamelet-binding-resource';
export * from './pipe-resource';
export * from './source-schema-config';
export * from './source-schema-type';
export { CAMEL_K_K8S_API_VERSION_V1 } from './camel-k-resource';
