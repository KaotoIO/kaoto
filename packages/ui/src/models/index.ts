export * from './camel-catalog-index';
export * from './camel-components-catalog';
export * from './camel-processors-catalog';
export * from './catalog-kind';
export * from './flow';
export * from './kamelets-catalog';
export * from './schema';
