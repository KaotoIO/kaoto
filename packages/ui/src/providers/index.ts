export * from './catalog-schema-loader.provider';
export * from './entities.provider';
