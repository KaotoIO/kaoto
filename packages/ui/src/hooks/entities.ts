import { useCallback, useMemo, useState } from 'react';
import { parse, stringify } from 'yaml';
import { CamelResource, SourceSchemaType, createCamelResource } from '../models/camel';
import { BaseCamelEntity } from '../models/camel/entities';
import { BaseVisualCamelEntity } from '../models/visualization/base-visual-entity';
import { EventNotifier } from '../utils';

export interface EntitiesContextResult {
  code: string;
  setCode: (code: string) => void;
  entities: BaseCamelEntity[];
  currentSchemaType: SourceSchemaType;
  setCurrentSchemaType: (entity: SourceSchemaType) => void;
  visualEntities: BaseVisualCamelEntity[];
  camelResource: CamelResource;
  updateCodeFromEntities: () => void;
  eventNotifier: EventNotifier;
}

export const useEntities = (): EntitiesContextResult => {
  const [sourceCode, setSourceCode] = useState<string>('');
  const eventNotifier = useMemo(() => new EventNotifier(), []);
  const [camelResource, setCamelResource] = useState<CamelResource>(createCamelResource());

  /** Set the Source Code and updates the Entities */
  const setCode = useCallback(
    (code: string) => {
      try {
        setSourceCode(code);
        const result = parse(code);
        const camelResource = createCamelResource(result);
        setCamelResource(camelResource);

        /** Notify subscribers that a `entities:update` happened */
        eventNotifier.next('entities:update', undefined);
      } catch (e) {
        setCamelResource(createCamelResource());
        console.error(e);
      }
    },
    [eventNotifier],
  );

  /** Updates the Source Code whenever the entities are updated */
  const updateCodeFromEntities = useCallback(() => {
    const code = stringify(camelResource) || '';
    setSourceCode(code);

    /** Notify subscribers that a `code:update` happened */
    eventNotifier.next('code:update', code);
  }, [camelResource, eventNotifier]);

  const setCurrentSchemaType = useCallback(() => {
    return (type: SourceSchemaType) => {
      setCamelResource(createCamelResource(type));
      updateCodeFromEntities();
    };
  }, [updateCodeFromEntities]);

  return useMemo(
    () => ({
      code: sourceCode,
      setCode,
      entities: camelResource.getEntities(),
      currentSchemaType: camelResource?.getType(),
      setCurrentSchemaType: setCurrentSchemaType(),
      visualEntities: camelResource.getVisualEntities(),
      camelResource,
      updateCodeFromEntities,
      eventNotifier,
    }),
    [sourceCode, setCode, setCurrentSchemaType, camelResource, updateCodeFromEntities, eventNotifier],
  );
};
