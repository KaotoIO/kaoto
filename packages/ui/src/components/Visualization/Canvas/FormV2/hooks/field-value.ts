import { useContext } from 'react';
import { safeGetValue } from '../../../../../utils';
import { ModelContext } from '../providers/ModelProvider';

export const useFieldValue = <T = unknown>(propertyPath: string) => {
  const { model, errors, onPropertyChange, disabled } = useContext(ModelContext);
  const propertyName = propertyPath.replace('#.', '');
  const value = safeGetValue(model, propertyName) as T | undefined;

  const propertyErrors = errors?.[propertyPath];

  const onChange = (value: T) => {
    onPropertyChange(propertyName, value);
  };

  return {
    value,
    errors: propertyErrors,
    onChange,
    disabled,
  };
};
