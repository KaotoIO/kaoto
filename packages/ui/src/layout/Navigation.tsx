import { Nav, NavExpandable, NavItem, NavList, PageSidebar, PageSidebarBody } from '@patternfly/react-core';
import clsx from 'clsx';
import { FunctionComponent, useContext, useMemo } from 'react';
import { Link, useLocation } from 'react-router-dom';
import { Links } from '../router/links.models';
import { NavElements } from './navigation.models';
import { EntitiesContext } from '../providers';
import { SourceSchemaType } from '../models/camel';
interface INavigationSidebar {
  isNavOpen: boolean;
}

export const Navigation: FunctionComponent<INavigationSidebar> = (props) => {
  const currentLocation = useLocation();
  const { currentSchemaType } = useContext(EntitiesContext)!;

  const navElements: NavElements = useMemo(
    () => [
      {
        title: 'Visualization',
        children: [
          { title: 'Design', to: Links.Home },
          { title: 'Source Code', to: Links.SourceCode },
        ],
      },
      {
        title: 'Beans',
        to: Links.Beans,
        hidden: () => !NAVIGATION_ELEMENTS.Beans.includes(currentSchemaType),
      },
      {
        title: 'Metadata',
        to: Links.Metadata,
        hidden: () => !NAVIGATION_ELEMENTS.Metadata.includes(currentSchemaType),
      },
      {
        title: 'Pipe ErrorHandler',
        to: Links.PipeErrorHandler,
        hidden: () => !NAVIGATION_ELEMENTS.PipeErrorHandler.includes(currentSchemaType),
      },
      {
        title: 'DataMapper',
        to: Links.DataMapper,
        hidden: () => !NAVIGATION_ELEMENTS.DataMapper.includes(currentSchemaType),
      },
      { title: 'Catalog', to: Links.Catalog },
    ],
    [currentSchemaType],
  );

  return (
    <PageSidebar isSidebarOpen={props.isNavOpen} id="vertical-sidebar">
      <PageSidebarBody>
        <Nav aria-label="Default global nav">
          <NavList>
            {navElements.map((nav, index) => {
              if ('children' in nav) {
                return (
                  <NavExpandable
                    id={nav.title}
                    key={nav.title}
                    data-testid={nav.title}
                    className={clsx({ 'pf-v6-u-hidden': nav.hidden?.() })}
                    hidden={nav.hidden?.()}
                    title={nav.title}
                    groupId={nav.title}
                    isActive={nav.children.some((child) => child.to === currentLocation.pathname)}
                    isExpanded
                  >
                    {nav.children.map((child) => (
                      <NavItem
                        id={child.title}
                        key={child.title}
                        data-testid={child.title}
                        itemId={index}
                        className={clsx({ 'pf-v6-u-hidden': child.hidden?.() })}
                        hidden={child.hidden?.()}
                        isActive={currentLocation.pathname === child.to}
                      >
                        <Link data-testid={child.title} to={child.to}>
                          {child.title}
                        </Link>
                      </NavItem>
                    ))}
                  </NavExpandable>
                );
              }

              return (
                <NavItem
                  id={nav.title}
                  className={clsx({ 'pf-v6-u-hidden': nav.hidden?.() })}
                  hidden={nav.hidden?.()}
                  key={nav.title}
                  data-testid={nav.title}
                  itemId={index}
                  isActive={currentLocation.pathname === nav.to}
                >
                  <Link data-testid={nav.title} to={nav.to}>
                    {nav.title}
                  </Link>
                </NavItem>
              );
            })}
          </NavList>
        </Nav>
      </PageSidebarBody>
    </PageSidebar>
  );
};

const NAVIGATION_ELEMENTS = {
  Beans: [SourceSchemaType.Route, SourceSchemaType.Kamelet],
  Metadata: [
    SourceSchemaType.Integration,
    SourceSchemaType.Kamelet,
    SourceSchemaType.KameletBinding,
    SourceSchemaType.Pipe,
  ],
  PipeErrorHandler: [SourceSchemaType.KameletBinding, SourceSchemaType.Pipe],
  DataMapper: [SourceSchemaType.Route, SourceSchemaType.Kamelet],
};
