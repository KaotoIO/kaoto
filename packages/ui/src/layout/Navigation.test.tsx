import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { EntitiesContextResult } from '../hooks/entities';
import { SourceSchemaType } from '../models/camel';
import { EntitiesContext } from '../providers';
import { Navigation } from './Navigation';

describe('Navigation Component', () => {
  it.each([
    SourceSchemaType.Route,
    SourceSchemaType.Kamelet,
    SourceSchemaType.Pipe,
    SourceSchemaType.KameletBinding,
    SourceSchemaType.Integration,
  ])('navigation sidebar for: %s', (currentSchemaType) => {
    const wrapper = render(
      <Router>
        <EntitiesContext.Provider value={{ currentSchemaType } as EntitiesContextResult}>
          <Navigation isNavOpen />
        </EntitiesContext.Provider>
      </Router>,
    );

    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
