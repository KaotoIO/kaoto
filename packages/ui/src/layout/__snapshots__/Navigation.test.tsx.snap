// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component navigation sidebar for: Integration 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    class="pf-v6-c-page__sidebar pf-m-expanded"
    id="vertical-sidebar"
  >
    <div
      class="pf-v6-c-page__sidebar-body"
    >
      <nav
        aria-label="Default global nav"
        class="pf-v6-c-nav"
        data-ouia-component-id="OUIA-Generated-Nav-5"
        data-ouia-component-type="PF6/Nav"
        data-ouia-safe="true"
      >
        <ul
          class="pf-v6-c-nav__list"
          role="list"
        >
          <li
            class="pf-v6-c-nav__item pf-m-expanded pf-m-current"
            data-ouia-component-id="OUIA-Generated-NavExpandable-5"
            data-ouia-component-type="PF6/NavExpandable"
            data-ouia-safe="true"
            data-testid="Visualization"
          >
            <button
              aria-expanded="true"
              class="pf-v6-c-nav__link"
              id="Visualization"
            >
              Visualization
              <span
                class="pf-v6-c-nav__toggle"
              >
                <span
                  class="pf-v6-c-nav__toggle-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v6-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </span>
              </span>
            </button>
            <section
              aria-labelledby="Visualization"
              class="pf-v6-c-nav__subnav"
            >
              <ul
                class="pf-v6-c-nav__list"
                role="list"
              >
                <li
                  class="pf-v6-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-29"
                  data-ouia-component-type="PF6/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    aria-current="page"
                    class="pf-v6-c-nav__link pf-m-current"
                    data-testid="Design"
                    href="/"
                  >
                    Design
                  </a>
                </li>
                <li
                  class="pf-v6-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-30"
                  data-ouia-component-type="PF6/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    class="pf-v6-c-nav__link"
                    data-testid="Source Code"
                    href="/code"
                  >
                    Source Code
                  </a>
                </li>
              </ul>
            </section>
          </li>
          <li
            class="pf-v6-c-nav__item pf-v6-u-hidden"
            data-ouia-component-id="OUIA-Generated-NavItem-31"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Beans"
              href="/beans"
            >
              Beans
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-32"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Metadata"
              href="/metadata"
            >
              Metadata
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item pf-v6-u-hidden"
            data-ouia-component-id="OUIA-Generated-NavItem-33"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Pipe ErrorHandler"
              href="/pipe-error-handler"
            >
              Pipe ErrorHandler
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item pf-v6-u-hidden"
            data-ouia-component-id="OUIA-Generated-NavItem-34"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="DataMapper"
              href="/datamapper"
            >
              DataMapper
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-35"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Catalog"
              href="/catalog"
            >
              Catalog
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Navigation Component navigation sidebar for: Kamelet 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    class="pf-v6-c-page__sidebar pf-m-expanded"
    id="vertical-sidebar"
  >
    <div
      class="pf-v6-c-page__sidebar-body"
    >
      <nav
        aria-label="Default global nav"
        class="pf-v6-c-nav"
        data-ouia-component-id="OUIA-Generated-Nav-2"
        data-ouia-component-type="PF6/Nav"
        data-ouia-safe="true"
      >
        <ul
          class="pf-v6-c-nav__list"
          role="list"
        >
          <li
            class="pf-v6-c-nav__item pf-m-expanded pf-m-current"
            data-ouia-component-id="OUIA-Generated-NavExpandable-2"
            data-ouia-component-type="PF6/NavExpandable"
            data-ouia-safe="true"
            data-testid="Visualization"
          >
            <button
              aria-expanded="true"
              class="pf-v6-c-nav__link"
              id="Visualization"
            >
              Visualization
              <span
                class="pf-v6-c-nav__toggle"
              >
                <span
                  class="pf-v6-c-nav__toggle-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v6-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </span>
              </span>
            </button>
            <section
              aria-labelledby="Visualization"
              class="pf-v6-c-nav__subnav"
            >
              <ul
                class="pf-v6-c-nav__list"
                role="list"
              >
                <li
                  class="pf-v6-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-8"
                  data-ouia-component-type="PF6/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    aria-current="page"
                    class="pf-v6-c-nav__link pf-m-current"
                    data-testid="Design"
                    href="/"
                  >
                    Design
                  </a>
                </li>
                <li
                  class="pf-v6-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-9"
                  data-ouia-component-type="PF6/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    class="pf-v6-c-nav__link"
                    data-testid="Source Code"
                    href="/code"
                  >
                    Source Code
                  </a>
                </li>
              </ul>
            </section>
          </li>
          <li
            class="pf-v6-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-10"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Beans"
              href="/beans"
            >
              Beans
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-11"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Metadata"
              href="/metadata"
            >
              Metadata
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item pf-v6-u-hidden"
            data-ouia-component-id="OUIA-Generated-NavItem-12"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Pipe ErrorHandler"
              href="/pipe-error-handler"
            >
              Pipe ErrorHandler
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-13"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="DataMapper"
              href="/datamapper"
            >
              DataMapper
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-14"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Catalog"
              href="/catalog"
            >
              Catalog
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Navigation Component navigation sidebar for: KameletBinding 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    class="pf-v6-c-page__sidebar pf-m-expanded"
    id="vertical-sidebar"
  >
    <div
      class="pf-v6-c-page__sidebar-body"
    >
      <nav
        aria-label="Default global nav"
        class="pf-v6-c-nav"
        data-ouia-component-id="OUIA-Generated-Nav-4"
        data-ouia-component-type="PF6/Nav"
        data-ouia-safe="true"
      >
        <ul
          class="pf-v6-c-nav__list"
          role="list"
        >
          <li
            class="pf-v6-c-nav__item pf-m-expanded pf-m-current"
            data-ouia-component-id="OUIA-Generated-NavExpandable-4"
            data-ouia-component-type="PF6/NavExpandable"
            data-ouia-safe="true"
            data-testid="Visualization"
          >
            <button
              aria-expanded="true"
              class="pf-v6-c-nav__link"
              id="Visualization"
            >
              Visualization
              <span
                class="pf-v6-c-nav__toggle"
              >
                <span
                  class="pf-v6-c-nav__toggle-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v6-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </span>
              </span>
            </button>
            <section
              aria-labelledby="Visualization"
              class="pf-v6-c-nav__subnav"
            >
              <ul
                class="pf-v6-c-nav__list"
                role="list"
              >
                <li
                  class="pf-v6-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-22"
                  data-ouia-component-type="PF6/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    aria-current="page"
                    class="pf-v6-c-nav__link pf-m-current"
                    data-testid="Design"
                    href="/"
                  >
                    Design
                  </a>
                </li>
                <li
                  class="pf-v6-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-23"
                  data-ouia-component-type="PF6/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    class="pf-v6-c-nav__link"
                    data-testid="Source Code"
                    href="/code"
                  >
                    Source Code
                  </a>
                </li>
              </ul>
            </section>
          </li>
          <li
            class="pf-v6-c-nav__item pf-v6-u-hidden"
            data-ouia-component-id="OUIA-Generated-NavItem-24"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Beans"
              href="/beans"
            >
              Beans
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-25"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Metadata"
              href="/metadata"
            >
              Metadata
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-26"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Pipe ErrorHandler"
              href="/pipe-error-handler"
            >
              Pipe ErrorHandler
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item pf-v6-u-hidden"
            data-ouia-component-id="OUIA-Generated-NavItem-27"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="DataMapper"
              href="/datamapper"
            >
              DataMapper
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-28"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Catalog"
              href="/catalog"
            >
              Catalog
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Navigation Component navigation sidebar for: Pipe 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    class="pf-v6-c-page__sidebar pf-m-expanded"
    id="vertical-sidebar"
  >
    <div
      class="pf-v6-c-page__sidebar-body"
    >
      <nav
        aria-label="Default global nav"
        class="pf-v6-c-nav"
        data-ouia-component-id="OUIA-Generated-Nav-3"
        data-ouia-component-type="PF6/Nav"
        data-ouia-safe="true"
      >
        <ul
          class="pf-v6-c-nav__list"
          role="list"
        >
          <li
            class="pf-v6-c-nav__item pf-m-expanded pf-m-current"
            data-ouia-component-id="OUIA-Generated-NavExpandable-3"
            data-ouia-component-type="PF6/NavExpandable"
            data-ouia-safe="true"
            data-testid="Visualization"
          >
            <button
              aria-expanded="true"
              class="pf-v6-c-nav__link"
              id="Visualization"
            >
              Visualization
              <span
                class="pf-v6-c-nav__toggle"
              >
                <span
                  class="pf-v6-c-nav__toggle-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v6-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </span>
              </span>
            </button>
            <section
              aria-labelledby="Visualization"
              class="pf-v6-c-nav__subnav"
            >
              <ul
                class="pf-v6-c-nav__list"
                role="list"
              >
                <li
                  class="pf-v6-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-15"
                  data-ouia-component-type="PF6/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    aria-current="page"
                    class="pf-v6-c-nav__link pf-m-current"
                    data-testid="Design"
                    href="/"
                  >
                    Design
                  </a>
                </li>
                <li
                  class="pf-v6-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-16"
                  data-ouia-component-type="PF6/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    class="pf-v6-c-nav__link"
                    data-testid="Source Code"
                    href="/code"
                  >
                    Source Code
                  </a>
                </li>
              </ul>
            </section>
          </li>
          <li
            class="pf-v6-c-nav__item pf-v6-u-hidden"
            data-ouia-component-id="OUIA-Generated-NavItem-17"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Beans"
              href="/beans"
            >
              Beans
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-18"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Metadata"
              href="/metadata"
            >
              Metadata
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-19"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Pipe ErrorHandler"
              href="/pipe-error-handler"
            >
              Pipe ErrorHandler
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item pf-v6-u-hidden"
            data-ouia-component-id="OUIA-Generated-NavItem-20"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="DataMapper"
              href="/datamapper"
            >
              DataMapper
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-21"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Catalog"
              href="/catalog"
            >
              Catalog
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Navigation Component navigation sidebar for: Route 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    class="pf-v6-c-page__sidebar pf-m-expanded"
    id="vertical-sidebar"
  >
    <div
      class="pf-v6-c-page__sidebar-body"
    >
      <nav
        aria-label="Default global nav"
        class="pf-v6-c-nav"
        data-ouia-component-id="OUIA-Generated-Nav-1"
        data-ouia-component-type="PF6/Nav"
        data-ouia-safe="true"
      >
        <ul
          class="pf-v6-c-nav__list"
          role="list"
        >
          <li
            class="pf-v6-c-nav__item pf-m-expanded pf-m-current"
            data-ouia-component-id="OUIA-Generated-NavExpandable-1"
            data-ouia-component-type="PF6/NavExpandable"
            data-ouia-safe="true"
            data-testid="Visualization"
          >
            <button
              aria-expanded="true"
              class="pf-v6-c-nav__link"
              id="Visualization"
            >
              Visualization
              <span
                class="pf-v6-c-nav__toggle"
              >
                <span
                  class="pf-v6-c-nav__toggle-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v6-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </span>
              </span>
            </button>
            <section
              aria-labelledby="Visualization"
              class="pf-v6-c-nav__subnav"
            >
              <ul
                class="pf-v6-c-nav__list"
                role="list"
              >
                <li
                  class="pf-v6-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-1"
                  data-ouia-component-type="PF6/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    aria-current="page"
                    class="pf-v6-c-nav__link pf-m-current"
                    data-testid="Design"
                    href="/"
                  >
                    Design
                  </a>
                </li>
                <li
                  class="pf-v6-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-2"
                  data-ouia-component-type="PF6/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    class="pf-v6-c-nav__link"
                    data-testid="Source Code"
                    href="/code"
                  >
                    Source Code
                  </a>
                </li>
              </ul>
            </section>
          </li>
          <li
            class="pf-v6-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-3"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Beans"
              href="/beans"
            >
              Beans
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item pf-v6-u-hidden"
            data-ouia-component-id="OUIA-Generated-NavItem-4"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Metadata"
              href="/metadata"
            >
              Metadata
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item pf-v6-u-hidden"
            data-ouia-component-id="OUIA-Generated-NavItem-5"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Pipe ErrorHandler"
              href="/pipe-error-handler"
            >
              Pipe ErrorHandler
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-6"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="DataMapper"
              href="/datamapper"
            >
              DataMapper
            </a>
          </li>
          <li
            class="pf-v6-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-7"
            data-ouia-component-type="PF6/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v6-c-nav__link"
              data-testid="Catalog"
              href="/catalog"
            >
              Catalog
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;
